packages/react-form-renderer/src/files/conditions-mapper.js

```diff
@@ -0,0 +1,89 @@
+/*
+  conditionsMapper will remap a conditions object and create an object with each depending fieldName as a key.
+
+  Since one field can be involed in more than one condition, an array of condition references will be created under each fieldName key
+
+  Since more than one field can be involved in the same condition, the same condition might be referenced from
+  several condition arrays.
+*/
+
+export const conditionsMapper = ({conditions}) => {
+  if (!conditions) return {};
+
+  function isObject(obj) {
+    return obj !== null && typeof obj === 'object' && !Array.isArray(obj);
+  }
+
+  function isArray(obj) {
+    return Array.isArray(obj);
+  }
+
+  function traverse({obj, fnc, key}) {
+    fnc && fnc({obj, key});
+
+    if (isArray(obj)) {
+      traverseArray({
+        obj,
+        fnc,
+        key,
+      });
+    } else if (isObject(obj)) {
+      traverseObject({
+        obj,
+        fnc,
+        key,
+      });
+    }
+  }
+
+  function traverseArray({obj, fnc, key}) {
+    for (var index = 0, len = obj.length; index < len; index++) {
+      const item = obj[index];
+      traverse({
+        obj: item,
+        fnc,
+        key: index,
+      });
+    }
+  }
+
+  function traverseObject({obj, fnc, key}) {
+    for (var index in obj) {
+      if (obj.hasOwnProperty(index)) {
```

Comment:
This check is redundant.  You are checking if an index from a set of object indexes is present inside the object.

You can use `Object.entries` in your traverse functions instead of the `for in` loop. That will give you both the index and value as params.

Reply:
I'll change to `entries` for better readability. But I do believe that the check for hasOwnProperty is needed in order to ensure that no inherited properties are traversed. This should not be the case with your schema object though.

Reply:
Fixed in 21361ec. I'm not sure how I should "resolve" these comments. Should I somehow link them to the commnit?